File: SDKs/digitalocean.com/2.0/go/pkg/models/operations/updatemaintenancewindow.go
package operations

import (
	"openapi/pkg/models/shared"
)

type UpdateMaintenanceWindowPathParams struct {
	DatabaseClusterUUID string `pathParam:"style=simple,explode=false,name=database_cluster_uuid"`
}

type UpdateMaintenanceWindowRequest struct {
	PathParams UpdateMaintenanceWindowPathParams
	Request    shared.Onev21databasesGetResponses200ContentApplication1jsonSchemaPropertiesDatabasesItemsPropertiesMaintenanceWindowAllOf0 `request:"mediaType=application/json"`
}

type UpdateMaintenanceWindow401ApplicationJSON struct {
	ID        string  `json:"id"`
	Message   string  `json:"message"`
	RequestID *string `json:"request_id"`
}

type UpdateMaintenanceWindowResponse struct {
	ContentType                                               string
	Headers                                                   map[string][]string
	StatusCode                                                int64
	UpdateMaintenanceWindow401ApplicationJSONObject           *UpdateMaintenanceWindow401ApplicationJSON
	Onev211ClicksGetResponses401ContentApplication1jsonSchema *shared.Onev211ClicksGetResponses401ContentApplication1jsonSchema
}

File: SDKs/digitalocean.com/2.0/typescript/src/sdk/models/operations/updatemaintenancewindow.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class UpdateMaintenanceWindowPathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=database_cluster_uuid" })
  databaseClusterUuid: string;
}


export class UpdateMaintenanceWindowRequest extends SpeakeasyBase {
  @Metadata()
  pathParams: UpdateMaintenanceWindowPathParams;

  @Metadata({ data: "request, media_type=application/json" })
  request: shared.Onev21databasesGetResponses200ContentApplication1jsonSchemaPropertiesDatabasesItemsPropertiesMaintenanceWindowAllOf0;
}


export class UpdateMaintenanceWindow401ApplicationJson extends SpeakeasyBase {
  @Metadata({ data: "json, name=id" })
  id: string;

  @Metadata({ data: "json, name=message" })
  message: string;

  @Metadata({ data: "json, name=request_id" })
  requestId?: string;
}


export class UpdateMaintenanceWindowResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  headers: Map<string, string[]>;

  @Metadata()
  statusCode: number;

  @Metadata()
  updateMaintenanceWindow401ApplicationJsonObject?: UpdateMaintenanceWindow401ApplicationJson;

  @Metadata()
  onev211ClicksGetResponses401ContentApplication1jsonSchema?: shared.Onev211ClicksGetResponses401ContentApplication1jsonSchema;
}
